frontend/src/app/shared/components/filtering-section/filtering-section.component.ts

```diff
@@ -1,21 +1,192 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { BaseComponent } from '@core/base/base.component';
+import { ChallengeService } from '@core/services/challenge.service';
+import { LanguageService } from '@core/services/language.service';
+import { TagService } from '@core/services/tag.service';
+import { ToastrNotificationsService } from '@core/services/toastr-notifications.service';
+import { ChallengeFilter } from '@shared/models/challenge/challenge-filter';
+import { ChallengePreview } from '@shared/models/challenge/challenge-preview';
+import { Language } from '@shared/models/language/language';
+import { PageSettings } from '@shared/models/page-settings';
+import { Tag } from '@shared/models/tag/tag';
+import { takeUntil } from 'rxjs';
 
 @Component({
     selector: 'app-filtering-section',
     templateUrl: './filtering-section.component.html',
     styleUrls: ['./filtering-section.component.sass'],
 })
-export class FilteringSectionComponent {
-    //TODO: Replace with real implementation
-    public challengesList = ['nothing', 'to', 'be'];
+export class FilteringSectionComponent extends BaseComponent implements OnInit {
+    public challenges: ChallengePreview[] = [];
 
-    //TODO: Replace with real implementation
-    public languagesList = ['All', 'HTML', 'C#', 'Java'];
+    public loading = false;
 
-    public statusesList = ['Approved & Beta', 'Approved', 'Beta'];
+    public languagesNames: string[] = [];
 
-    public progressesList = ['All', 'Not completed', 'Completed', 'Not trained on'];
+    public statusesNames: string[] = [];
 
-    //TODO: Replace with real implementation
-    public tagsList = ['All', 'Algorithms', 'Strings', 'Data Types', 'Formatting', 'Logic'];
+    public progressesNames: string[] = [];
+
+    public tagsNames: string[] = [];
+
+    private page: PageSettings = {
+        pageNumber: 0,
+        pageSize: 10,
+    };
+
+    private isLastPage = false;
+
+    private filter: ChallengeFilter = {
+        title: '',
+        tagsIds: [],
+    };
+
+    private progresses = [
+        { name: 'All' },
+        { name: 'Not completed', id: 1 },
+        { name: 'Completed', id: 2 },
+        { name: 'Not trained on', id: 0 },
+    ];
+
+    private tags: Tag[] = [];
+
+    private languages: Language[] = [];
+
+    private statuses = [
+        { name: 'Approved & Beta' },
+        { name: 'Approved', id: 2 },
+        { name: 'Beta', id: 0 },
+    ];
```

Comment:
store all these data in a separate file filtering-section.utils.ts (in the same folder as component)

Reply:
done